CONTRIBUTING.md

```diff
@@ -0,0 +1,270 @@
+packages, there are two critical pieces of information you need to add to the
+commit body:
+
+- A brief explanation as to why this re-vendor is required.
+
+  For example, it might be that an important bug fix or new feature is
+  required. It might even be that a particular commit is needed. If so, please
+  state that.
+
+- The range of commits being added for these third-party packages.
+
+  It is possible that re-vendoring a particular package will also result in updates to
+  other dependent packages. However, it is important to include the commit range
+  (even if it is a big) for the primary package(s) the re-vendor PR has been
+  raised for.
+
+  These details allow for easier problem-determination in the future should
+  the re-vendor PR introduce bugs or behavioural changes.
+
+  Generate the list of new commits added to the primary re-vendored
+  package by comparing the previous and latest commits for the package being re-vendored.
+
+  For example, if a new version of `libcontainer` (part of the `runc`)
+  repository is required:
+
+  1. Determine the previous and latest commits for the package by looking at
+     the `diff` of the `Gopkg.toml` file in your branch.
+
+  1. Run the commands below:
+
+     ```
+     $ cd $GOPATH/src/github.com/opencontainers/runc
+     $ old_commit="..."
+     $ new_commit="..."
+     $ git log --no-merges --abbrev-commit --pretty=oneline "${old_commit}..${new_commit}") | sed 's/^/    /g'
+     ```
+
+  The output of the last command above should be pasted directly into the
+  commit "as-is". Note that the four space indent added by the `sed` command is
+  used to force github to render the list in a fixed-width font which makes the
+  list easier to read.
+
+## Patch format
+
+Beside the `Signed-off-by` footer, we expect each patch to comply with the
+following format:
+
+```
+Subsystem: Change summary (no longer than 75 characters)
+
+More detailed explanation of your changes: Why and how.
+Wrap it to 72 characters.
+
+See:
+    http://chris.beams.io/posts/git-commit/
+
+For some more good advice, and the Linux Kernel document:
+    https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/SubmittingPatches
+
+Fixes: #nnn
+
+Signed-off-by: <contributor@foo.com>
+```
+
+For example:
+
+```
+pod: Remove token from Cmd structure
+
+The token and pid data will be hold by the new Process structure and
+they are related to a container.
+
+Fixes: #123
+
+Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>
+```
+
+Correct formatting of the PR patches is verified using the
+[checkcommits](https://github.com/kata-containers/tests/tree/master/cmd/checkcommits)
+tool.
+
+Note, that the body of the message should not just be a continuation of the
+subject line, and is not used to extend the subject line beyond its length
+limit. They should stand alone as complete sentence and paragraphs.
+
+It is recommended that each of your patches fixes one thing. Smaller patches
```

Comment:
ah, and here is the 'small patches' bit - so above we probably need to say 'the below' or similar when referencing it - or just 'elsewhere in this document' etc.